apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockImpl.java

```diff
@@ -188,15 +188,23 @@ public int getPayloadLength() {
     }
 
     @Override
-    public List<TransactionImpl> getTransactions() {
+    public List<Transaction> getTransactions() {
```

Comment:
Did you consider this method calls, which relies on loading transactions from db?

Reply:
It has replacement code for filling in those Tr data by using separate transactionDb call.

See lines 1635 - 1655  in BlockchainProcessorImpl.java
methods:
public List<? extends Transaction> loadTransactions(Block block) {
public List<? extends Transaction> getTransactions(Block block) {

Reply:
Generator:103 - `block.getTransactions()` will return actual list of transactions?

Reply:
code is reverted to using BlockImpl.loadTransactions() in three places BlockchainProcessorImpl.java.